### PR TITLE
Adding TTL 7487

### LINKED ISSUE #1682: Adding TTL 7487
Short description: 4-bit True/complement, zero/one 
This would be a good 74 series chip for use behind an adder to set up for add/subtract and +1/-1 operations
Link to datasheet: [https://archive.org/details/bitsavers_tidataBookVol2_45945352/page/n403/mode/2up](https://archive.org/details/bitsavers_tidataBookVol2_45945352/page/n403/mode/2up)

I currently intend to add this myself, as it seems like an easy first part.

### PR DESCRIPTION
**Pull request by aplq**
This pull request is to close #1682

### CODE REVIEW

File: src/main/resources/resources/logisim/strings/std/std.properties [hunk 1]
```diff
@@ -819,6 +819,7 @@ TTL747266 = 747266: quad 2-input XNOR gate
 TTL7474 = 7474: dual D-Flipflops with preset and clear
 TTL7485 = 7485: 4-bit magnitude comparator
 TTL7486 = 7486: quad 2-input XOR gate
+TTL7487 = 7487: 4-bit True/complement, zero/one
```
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std.properties:822]: Please add commented-out placeholder for this localisation string to all other supported langues.

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	public static final byte Y4=8;
+	
+	//No Connect (datasheet pins)
+	public static final byte NC1=4;
+	public static final byte NC2=11;
+	
+	
+	//private static final byte pinCount = 14;
+	//private static final byte[] outPorts = {Y1, Y2, Y3+1, Y4+1};
+	//private static final byte[] ncPorts = {NC1, NC2+1};
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java:47]: Remove commented out code

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * This is free software released under GNU GPLv3 license
+ */
+
+package com.cburch.logisim.std.ttl;
+
+import com.cburch.logisim.data.Value;
+import com.cburch.logisim.instance.InstancePainter;
+import com.cburch.logisim.instance.InstanceState;
+
+public class Ttl7487 extends AbstractTtlGate {
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java:16]: Please add a documentation comment to the class describing its purpose and referencing the TTL part's data sheet.

File: CHANGES.md [hunk 4]
```diff
@@ -7,6 +7,7 @@
   * Fixed Karnaugh map color index bug.
   * Attribute sheet now honors application color theme.
   * Attribute sheet now displays HEX value of color properties.
+  * ADDED TTL 7487: 7487: 4-bit True/complement, zero/one
```
Comment (maehne) [inline CHANGES.md:10]: "ADDED" -> "Added"

**Review by maehne — CHANGES_REQUESTED** [review]
@aplq: Looks good except for details.

@BunchOfBits: Could you please have a look, too? You have have been contributing most to the TTL library, which is outside of our focus.

**Comment by BunchOfBits** [comment]
@maehne I don't have review authorization, so I can't comment on the code directly, but I think you have already touched all important points, apart from the code style violations which is a blocking check. The logic in propagateTtl() could be slightly more compact by using a single 4-bit Value object instead of 4 1-bit Value objects, but that is maybe a matter of preference, the pull request will work just fine either way.

You could consider the following reference to the datasheet in the comments:

Page 3-354 of http://bitsavers.org/components/ti/_dataBooks/1985_TI_The_TTL_Data_Book_Vol_2.pdf

**Comment by maehne** [comment]
@BunchOfBits: Thanks for your feedback on this PR! I agree with your recommendations.

**Comment by maehne** [comment]
@aplq: What is the status of your PR? Could you please address our feedback and resolve the conflict in `CHANGES.md` so that this PR can move forward?

File: src/main/resources/resources/logisim/strings/std/std.properties (hunk 1 above)
Comment (aplq) [reply src/main/resources/resources/logisim/strings/std/std.properties:822]: Added in recent commit

File: CHANGES.md (hunk 4 above)
Comment (aplq) [reply CHANGES.md]: Changed in recent commit

**Comment by aplq** [comment]
@maehne I believe all changes are made and conflicts resolved

File: src/main/resources/resources/logisim/strings/std/std.properties (hunk 1 above)
Comment (maehne) [reply src/main/resources/resources/logisim/strings/std/std.properties:822]: The component name is incomplete according to the [data sheet](https://archive.org/details/bitsavers_tidataBookVol2_45945352/page/n403/mode/2up). Please, use "4-bit True/complement, zero/one elements"

File: src/main/resources/resources/logisim/strings/std/std_de.properties [hunk 5]
```diff
@@ -818,6 +818,7 @@ TTL747266 = 747266: Vierfach 2-Eingangs-XNOR-Gatter
 TTL7474 = 7474: Zweifach D-Flipflops mit Preset und Clear
 TTL7485 = 7485: 4-Bit Größenvergleicher
 TTL7486 = 7486: Vierfach 2-Eingangs-XOR-Gatter
+# ==> TTL7487 =
```
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_de.properties:821]: German translation: "7487: 4-Bit Wahr/Komplement, Null/Eins-Elemente"

File: src/main/resources/resources/logisim/strings/std/std_fr.properties [hunk 6]
```diff
@@ -819,6 +819,7 @@ TTL747266 = 747266 : quadruple portes XNOR 2 entrées
 TTL7474 = 7474 : deux bascules D prépositionnables et réinitialisables
 TTL7485 = 7485 : comparateur de magnitude 4 bits
 TTL7486 = 7486 : quadruple portes OU exclusif (XOR) à 2 entrées
+# ==> TTL7487 =
```
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_fr.properties:822]: French translation: "7487 : 4 bits Vrai/complément, éléments zéro/un"

File: CHANGES.md [hunk 7]
```diff
@@ -19,6 +19,7 @@
   * Fixed Karnaugh map color index bug.
   * Attribute sheet now honors application color theme.
   * Attribute sheet now displays HEX value of color properties.
+  * Added TTL 7487: 7487: 4-bit True/complement, zero/one
```
Comment (maehne) [inline CHANGES.md:22]: Use full component name "7487: 4-bit True/complement, zero/one elements"

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+import com.cburch.logisim.instance.InstanceState;
+
+/*
+ * Class simulates a TTL7487 IC
+ * It takes a 4 bit input (A1-A4) and performs one of the following oppertations based on inputs B and c to produce outputs Y1-Y4
+ *   B=0, C=0 output compliment of A
+ *   B=0, C=1 output A unchanged
+ *   B=1, C=0 output all High
+ *   B=1, C=1 output all Low
+ * Likely use is before an adder to give control for addition/subtraction along with ++,--,transparent in conjunction with the adder's carry in bit
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java:23]: Do you mean with "++" and "--" increment and decrement, respectively?

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+package com.cburch.logisim.std.ttl;
+
+import com.cburch.logisim.data.Value;
+import com.cburch.logisim.instance.InstancePainter;
+import com.cburch.logisim.instance.InstanceState;
+
+/*
+ * Class simulates a TTL7487 IC
+ * It takes a 4 bit input (A1-A4) and performs one of the following oppertations based on inputs B and c to produce outputs Y1-Y4
+ *   B=0, C=0 output compliment of A
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java:19]: Typo: "compliment" -> "complement"

**Review by maehne — CHANGES_REQUESTED** [review]
Sorry for the delay! Please merge our `main` branch before resolving the remaining small issues.

File: src/main/resources/resources/logisim/strings/std/std_fr.properties (hunk 6 above)
Comment (aplq) [reply src/main/resources/resources/logisim/strings/std/std_fr.properties]: Added in commit

File: CHANGES.md (hunk 7 above)
Comment (aplq) [reply CHANGES.md]: Fixed in commit

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java (hunk 8 above)
Comment (aplq) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl7487.java]: I believed it is clearer now

File: src/main/resources/resources/logisim/strings/std/std.properties (hunk 1 above)
Comment (maehne) [reply src/main/resources/resources/logisim/strings/std/std.properties:837]: The string is still incomplete, as "elements" is missing.

**Review by maehne — CHANGES_REQUESTED** [review]

File: src/main/resources/resources/logisim/strings/std/std.properties (hunk 1 above)
Comment (aplq) [reply src/main/resources/resources/logisim/strings/std/std.properties]: Sorry about that it should be fixed now

**Review by maehne — APPROVED** [review]

**Pull request merged**